std/test/__init__.py

```diff
@@ -0,0 +1,619 @@
+
+        # Update best
+        if len(self.best_fit) < 2:
+            self.best_fit = self.best_fit + [coin]
+        else:
+            cur_best_fit_amount = sum(map(lambda x: x.amount, self.best_fit))
+            greater = 0 if self.best_fit[0].amount > self.best_fit[1].amount else 1
+            lesser = (greater + 1) % 2
+            candidate_bf_amount = self.best_fit[lesser].amount + coin.amount
+            if candidate_bf_amount < cur_best_fit_amount and candidate_bf_amount > self.target:
+                self.best_fit = [self.best_fit[lesser], coin]
+
+# A basic wallet that knows about standard coins.
+# We can use this to track our balance as an end user and keep track of chia
+# that is released by contracts, if the contracts interact meaningfully with
+# them, as many likely will.
+class Wallet:
+    def __init__(self,parent,name,pk,priv):
+        """Internal use constructor, use Network::make_wallet
+
+        Fields:
+        parent - The Network object that created this Wallet
+        name - The textural name of the actor
+        pk_ - The actor's public key
+        priv_ - The actor's private key
+        usable_coins - Standard coins spendable by this actor
+        puzzle - A program for creating this actor's standard coin
+        puzzle_hash - The puzzle hash for this actor's standard coin
+        """
+        self.parent = parent
+        self.name = name
+        self.pk_ = pk
+        self.priv_ = priv
+        self.usable_coins = {}
+        self.puzzle = puzzle_for_pk(self.pk())
+        self.puzzle_hash = self.puzzle.get_tree_hash()
+
+    def __repr__(self):
+        return f'<Wallet(name={self.name},puzzle_hash={self.puzzle_hash},pk={self.pk_})>'
+
+    # Make this coin available to the user it goes with.
+    def add_coin(self,coin):
+        self.usable_coins[coin.name()] = coin
+
+    def compute_combine_action(self,amt,actions,usable_coins):
+        # No point in going on if we're on our last dollar.
+        if len(usable_coins) < 2:
+            return None
+
+        # No one coin is enough, try to find a best fit pair, otherwise combine the two
+        # maximum coins.
+        searcher = CoinPairSearch(amt)
+
+        # Process coins for this round.
+        for k,c in usable_coins.items():
+            searcher.process_coin_for_combine_search(c)
+
+        best_fit, max_coins = searcher.get_result()
+
+        # Best fit coins indicate we can combine just 2, so do that.
+        if len(best_fit) == 2:
+            return actions + [best_fit]
+        else:
+            del usable_coins[max_coins[0].name()]
+            del usable_coins[max_coins[1].name()]
+            new_coin = CoinWrapper(
+                max_coins[0],
+                self.puzzle_hash,
+                max_coins[0].amount + max_coins[1].amount,
+                self.puzzle
+            )
+            usable_coins[new_coin.name()] = new_coin
+            return compute_combine_action(amt, actions + [max_coins], usable_coins)
```

Comment:
yes true.  I did it this way because it was followable and a bit more eyeball comprehensible.  I'll switch it to aggregating a best fit collection from all coins.

Reply:
now combines more coins at once.